core/src/main/java/cucumber/runtime/LinePredicate.java

```diff
@@ -0,0 +1,31 @@
+package cucumber.runtime;
+
+import gherkin.events.PickleEvent;
+import gherkin.pickles.PickleLocation;
+
+import java.util.List;
+import java.util.Map;
+
+public class LinePredicate implements PicklePredicate {
+    private Map<String, List<Long>> lineFilters;
+
+    public LinePredicate(Map<String, List<Long>> lineFilters) {
+        this.lineFilters = lineFilters;
+    }
+
+    @Override
+    public boolean apply(PickleEvent pickleEvent) {
+        String picklePath = pickleEvent.uri;
+        if (!lineFilters.containsKey(picklePath)) {
+            return true;
```

Comment:
That makes sense. The runner presents each file[:line] arguments as having or semantics. But internally it first finds all feature files by looking at the file part and then reduces the set of tests with the line predicate.